src/core/ddsi/src/ddsi_proxy_participant.c

```diff
@@ -282,15 +235,13 @@ void ddsi_proxy_participant_add_pwr_lease_locked (struct ddsi_proxy_participant
   ddsrt_fibheap_insert (&ddsi_lease_fhdef_pp, lh, pwr->lease);
   minl_new = ddsrt_fibheap_min (&ddsi_lease_fhdef_pp, lh);
   /* ensure proxypp->minl_man/minl_auto is equivalent to min(leaseheap_man/auto) */
-  if (proxypp->owns_lease && minl_prev != minl_new)
+  if (minl_prev != minl_new)
   {
     ddsrt_etime_t texp = ddsrt_etime_add_duration (ddsrt_time_elapsed (), minl_new->tdur);
     struct ddsi_lease *lnew = ddsi_lease_new (texp, minl_new->tdur, minl_new->entity);
     if (minl_prev == NULL)
     {
-      assert (manbypp);
-      assert (ddsrt_atomic_ldvoidp (&proxypp->minl_man) == NULL);
-      ddsrt_atomic_stvoidp (&proxypp->minl_man, lnew);
+      ddsrt_atomic_stvoidp (manbypp ? &proxypp->minl_man : &proxypp->minl_auto, lnew);
```

Comment:
`minl_auto` always has a lease set (in `ddsi_new_proxy_participant`), so I think the asserts that were here are still valid and this path should never store a lease in `proxypp->minl_auto`?

Reply:
I think you're right: it used to be that lease was present only if `owns_lease` was set, but this PR changes that. That I removed the asserts is not because of that change, but because I only later decided to remove all that code for `owns_lease` = `false` hence had to remove them, and then forgot to put them back.

I'll have another look just to be sure 😀